### ISSUE TITLE
Missiles per target functionality enhancement

### ISSUE THREAD

**Issue by jrodrigv** [PapaJoesSoup/BDArmory#393]
##### ISSUE TYPE
 - Enhancement  (Expand on existing features)

##### KSP and BDA Version
DEV version of BDA

##### SUMMARY
Missiles per target functionality is not working as it should.

Currently, is not taking into account the missiles flying per target but all the missiles fired.

Weapon manager module should consider the missiles per target and choose a different target if that one already has the max number of missiles locked on it.

##### STEPS TO REPRODUCE

- Add just 6 AAM-120 missiles to plane and select max 2 missiles/target.
- Add 4 passive enemies.
- Activate Guard mode and Autopilot on the plane with the 6 AMM-120 missiles.
- You will see how 2 missiles are fired to the same target, and then it will stop doing anything else. It will not select any new targets, it will not fire any other missiles until at least on the missiles has detonated or missed.

**Comment by MrSteve10424** [comment]
True, I think its simply a ksp limitation though, don't quote me because im not that good at coding.
